File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.cpp
class Solution {
public:
    string largestOddNumber(string s) {
        string ans;
        int n=s.length();
        int check_point=0;
        for(int i=0;i<n;i++)
        {
            int num=s[i]-'0';
            if(num%2==1)
            {
                for(int j=check_point;j<=i;j++)
                    ans.push_back(s[j]);
                check_point=i+1;
            }
            

        }

        return ans;
        
    }
};

File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String a) {
        int index=-1;
        for(int i=a.length()-1;i>=0;i--)
        {
            int n=(int)(a.charAt(i));
            if(n%2==1)
            {
                    index=i;
                    break;
            }
        }
        StringBuilder ans=new StringBuilder();
        if(index==-1)
            return ans.toString();
        for(int i=0;i<=index;i++)
        {
            ans.append(a.charAt(i));
        }
        return ans.toString();
        
    }
}